src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,683 @@
+            return compare(version1, version2) < 0;
+        }
+    }
+    
+    private static String extractVersion(String fileName) {
+        int startIndex = INSTALL_MAP_PREFIX.length() + 1;
+        int endIndex = fileName.lastIndexOf(INSTALL_MAP_SUFFIX);
+        if (startIndex < endIndex) {
+            return fileName.substring(startIndex, endIndex);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Performs pairwise comparison of version strings, including nulls and non-integer components.
+     * @param version1
+     * @param version2
+     * @return positive if version2 is greater, negative if version1 is greater, otherwise 0
+     */
+    private static int compare(String version1, String version2) {
+        if (version1 == null && version2 == null) {
+            return 0;
+        } else if (version1 == null && version2 != null) {
+            return -1;
+        } else if (version1 != null && version2 == null) {
+            return 1;
+        }
+        String[] components1 = version1.split("\\.");
+        String[] components2 = version2.split("\\.");
+        for (int i = 0; i < components1.length && i < components2.length; i++) {
+            int comparison;
+            try {
+                comparison = new Integer(components1[i]).compareTo(new Integer(components2[i]));
+            } catch (NumberFormatException e) {
+                comparison = components1[i].compareTo(components2[i]);
+            }
+            if (comparison != 0) {
+                return comparison;
+            }
+        }
+        return components1.length - components2.length;
+    }
+    
+    /**
+     * Performs product validation by running bin/productInfo validate
+     * 
+     * @throws PluginExecutionException
+     *             if product validation failed or could not be run
+     */
+    public void productInfoValidate() throws PluginExecutionException {
+        Process pr = null;
+        InputStream is = null;
+        Scanner s = null;
+        Worker worker = null;
+        try {
+            String osName = System.getProperty("os.name", "unknown").toLowerCase();
+            boolean isWindows = osName.indexOf("windows") >= 0;
+            String command;
+            if (isWindows) {
+                command = installDirectory + "\\bin\\productInfo.bat validate";
+            } else {
+                command = installDirectory + "/bin/productInfo validate";
+            }
+            pr = Runtime.getRuntime().exec(command);
+            worker = new Worker(pr);
+            worker.start();
+            worker.join(300000);
+            if (worker.exit == null) {
+                throw new PluginExecutionException("Product validation error: timeout");
+            }
+            int exitValue = pr.exitValue();
+            if (exitValue != 0) {
+                is = pr.getInputStream();
+                s = new Scanner(is);
+                s.useDelimiter("\\A");
```

Comment:
What is this delimeter?

Reply:
Regex to match beginning of input.  Added a comment to clarify.